aten/src/ATen/autocast_mode.cpp

```diff
@@ -91,6 +112,30 @@ Tensor cached_cast(at::ScalarType to_type, const Tensor& arg) {
   }
 }
 
+// Overload cached_cast with extra parameter DeviceType
+Tensor cached_cast(DeviceType device_type, at::ScalarType to_type, const Tensor& arg) {
```

Comment:
I guess this is fine for BC reasons, but you should at least make the original overload call into this overload to reuse code!

Reply:
Thanks for the suggestion. How about we change the `cached_cast` into `TORCH_API Tensor cached_cast(at::ScalarType to_type, const Tensor& arg, DeviceType device_type=DeviceType::CUDA);` with default device parameter of CUDA? Then  `cached_cast` is back compatible and also can be used for AutocastCPU.

Reply:
That also is very reasonable.

Reply:
Thanks~

Reply:
Oops, It seems we don't need to calculate the dispatchkey for tensor. Instead we can directly use `bool is_enabled()` or `bool is_cpu_enabled()` to distinguish whether AutocastCUDA or AutocastCPU is enabled. Then we can also save the argument of `device_type`.

Reply:
Emmm... Because we use the `ExcludeDispatchKeyGuard`, so `is_cuda_enabled()` or `is_cpu_enabled()` will always return false. Revert back to the implementation with argument of `device_type`.